js/apps/admin-ui/src/clients/authorization/ResourcesPolicySelect.tsx

```diff
@@ -108,57 +128,133 @@ export const ResourcesPolicySelect = ({
     [search]
   );
 
+  const [toggleUnsavedChangesDialog, UnsavedChangesConfirm] = useConfirmDialog({
+    titleKey: t("common:unsavedChangesTitle"),
+    messageKey: t("common:unsavedChangesConfirm"),
+    continueButtonLabel: t("common:continue"),
+    continueButtonVariant: ButtonVariant.danger,
+    onConfirm: async () => {
+      navigate(
+        toPolicyDetails({
+          realm: realm,
+          id: clientId,
+          policyId: clickedPolicy?.id!,
+          policyType: clickedPolicy?.type!,
+        })
+      );
+    },
+  });
+
   const toSelectOptions = () =>
     items.map((p) => (
       <SelectOption key={p.id} value={p.id}>
         {p.name}
       </SelectOption>
     ));
 
+  const toChipGroupItems = (
+    field: ControllerRenderProps<PolicyRepresentation, Type>
+  ) => {
+    return (
+      <ChipGroup>
+        {(field.value as string[]).map((permissionId) => {
+          const policy = items.find(
+            (permission) => permission.id === permissionId
+          );
+          if (policy) {
+            return (
+              <Chip
+                key={policy.id}
+                onClick={(e: React.MouseEvent) => {
+                  e.stopPropagation();
```

Comment:
There should be no need to add the type here, it can be inferred from the signature of the prop. Also, let's avoud single letter variable names where possible.
```suggestion
                onClick={(event) => {
                  event.stopPropagation();
```